src/library/fixitem/FixItemControl.java

```diff
@@ -0,0 +1,72 @@
+package library.fixitem;
+import library.entities.Item;
+import library.entities.Library;
+
+public class FixItemControl {
+    
+    private enum ControlState { INITIALISED, READY, INSPECTING };
+    private ControlState state;
+    private FixItemUI ui;
+    
+    private Library library;
+    private Item currentItem;
+
+
+    public FixItemControl() {
+        this.library = Library.getInstance();
+        state = ControlState.INITIALISED;
+    }
+    
+    
+    public void setUI(FixItemUI ui) {
+        if (!state.equals(ControlState.INITIALISED))
+            throw new RuntimeException("FixItemControl: cannot call setUI except in INITIALISED state");
+            
+        this.ui = ui;
+        this.ui.SeTrEaDy();
+        state = ControlState.READY;
+    }
+
+
+    public void itemScanned(long itemId) {
+        if (!state.equals(ControlState.READY))
+            throw new RuntimeException("FixItemControl: cannot call itemScanned except in READY state");
```

Comment:
PR 46 D2 CSG#7 Curly brackets must be used.

Reply:
[efb76bd](https://github.com/elliotmitchum/ITC205/pull/46/commits/efb76bdba38c1481c2c5f01f19f5323ccc95d3c6)